inspirehep/modules/disambiguation/logic.py

```diff
@@ -0,0 +1,438 @@
+    author profile).
+
+    After clustering and matching against existing clusters (if any),
+    this method is deciding how the cluster should be processed.
+
+    If the cluster is new (ie. new author, no profile), the method is
+    dispatching a job to create a new profile and assigning the given cluster
+    to this profile.
+
+    If the cluster was matched with existing cluster (signatures pointing to
+    the same author profile), then all signatures will be overwritten to point
+    to the profile (recid_key parameter).
+
+    In case if an author has claimed her or his paper (ground truth),
+    then it is known that the author has the profile already.
+    This profile is overwriting all signatures within the same cluster.
+
+    If the clustering task returned more than two claimed signatures,
+    belonging to different authors, then re-clustering is triggered.
+    Each of the claimed signatures become a bucket and each not claimed
+    one is being allocated to one of the buckets based on the likelihood
+    of representing the same author.
+
+    :param uuids:
+        A list of signatures, representing the same author.
+
+        Example:
+            uuids = ['4156520-560d-4248-a57f-949c361e0dd0']
+
+    :param signatures:
+        A list of signatures collected during clustering.
+
+        Example:
+            signatures =
+                [{'author_affiliation': u'MIT, Cambridge, CTP',
+                  'author_name': u'Wang, Yi-Nan',
+                  'publication_id': u'9d3dca5d-3551-4ca4-9b52-63db656e4793',
+                  'signature_id': u'a4156520-560d-4248-a57f-949c361e0dd0',
+                  'author_recid': u'10123',
+                  'author_claimed: False}]
+
+    :param recid_key:
+        A record id representing profile, which 'old' cluster is
+        associated with.
+
+        Example:
+            recid_key = '10123'
+    """
+    # Create a map where each signature can be accessed by its uuid.
+    signatures_map = _create_uuid_key_dictionary(signatures)
+
+    # Count claimed signatures. Set allows for unique signatures.
+    claims = set()
+
+    for uuid in uuids:
+        claim_status = _check_if_claimed(signatures_map[uuid])
+
+        # If it is claimed, and profile_id is indeed a digit.
+        if claim_status[0] and claim_status[1].isdigit():
+            claims.add(claim_status)
+
+    # If there are no claimed signatures and no match with an 'old' cluster.
+    if len(claims) == 0 and not recid_key:
+        # Select the most rich-in-information signature.
+        base_profile = get_signature(_select_profile_base(signatures_map,
+                                                          uuids))
+        # Create a new profile.
+        recid = create_author(base_profile)
+
+        # Update all signatures with the new profile (recid).
+        for uuid in uuids:
+            record = signatures_map[uuid].get('publication_id')
+            celery.current_app.send_task(
```

Comment:
I have some problems with imports when I do
```python
from inspirehep.modules.disambiguation.tasks import update_authors_recid
```
in `logic.py`, my `Celery` worker is crashing.